File: packages/swift/Sources/CatalystSwift/Models/EpiclesePlatform.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

import Foundation

/// A platform recognized by Epiclese (e.g. VRChat, Resonite).
public struct EpiclesePlatform: Decodable, Sendable, Equatable, Hashable {
  public let id: String
  public let name: String
  public let description: String
  public let url: String?
  public let startAt: Date?
}

public struct EpiclesePlatformWrapper: Decodable, Sendable {
  public let platform: EpiclesePlatform
}

public struct EpiclesePlatformsWrapper: Decodable, Sendable {
  public let platforms: [EpiclesePlatform]
}

File: packages/kotlin/src/main/kotlin/com/natsuneko/catalyst/models/EpiclesePlatform.kt
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

package com.natsuneko.catalyst.models

import kotlinx.serialization.Serializable
import kotlin.time.ExperimentalTime

/**
 * A platform recognized by Epiclese (e.g. VRChat, Resonite)
 */
@Serializable
data class EpiclesePlatform @OptIn(ExperimentalTime::class) constructor(
    val id: String,
    val name: String,
    val description: String,
    val url: String? = null,
    val startAt: kotlin.time.Instant? = null
)

/**
 * Wrapper for the `{"platform": {...}}` response shape
 */
@Serializable
internal data class EpiclesePlatformWrapper(val platform: EpiclesePlatform)

/**
 * Wrapper for the `{"platforms": [...]}` response shape
 */
@Serializable
internal data class EpiclesePlatformsWrapper(val platforms: List<EpiclesePlatform>)
